### PR TITLE
[luci-interpreter/loader] Support CircleReferencingConst load

### PR DESCRIPTION
**Pull request by m-bronnikov**
This commit adds support of CircleReferencingConst custom node to loader.

ONE-DCO-1.0-Signed-off-by: Maksim Bronnikov <max120199@gmail.com>

------------------------

For: #8042
Draft: #8190

### CODE REVIEW

**Comment by m-bronnikov** [comment]
This PR conflicts a bit with #8318. Let's wait until #8318 is merged and rebase this PR after that.

**Comment by seanshpark** [comment]
@m-bronnikov , #8318 landed, please rebase.

File: compiler/luci-interpreter/src/loader/GraphLoader.cpp
Comment (seanshpark) [inline compiler/luci-interpreter/src/loader/GraphLoader.cpp:121]: If we throw here, maybe also check and throw inside `isTensorProducingNode()` for consistancy?

Comment (m-bronnikov) [reply compiler/luci-interpreter/src/loader/GraphLoader.cpp:121]: Hmm, this suggestion makes sense, but I don't like idea of adding more checks for supported `CircleCustom` cases. So, I think it would be better to add another helper: `isValidCustomNode` and check Custom node is valid only once in thi s helper.

Comment (m-bronnikov) [reply compiler/luci-interpreter/src/loader/GraphLoader.cpp:121]: And therefore remove this throw here.

Comment (m-bronnikov) [reply compiler/luci-interpreter/src/loader/GraphLoader.cpp]: > So, I think it would be better to add another helper: `isValidCustomNode` and check Custom node is valid only once in thi s helper.

I added this check with name "isSupportedCustomNode".

File: compiler/luci-interpreter/src/loader/GraphLoader.cpp
Comment (seanshpark) [inline compiler/luci-interpreter/src/loader/GraphLoader.cpp:86]: (optional) I assume this struct exist also in other file. So I think it would be better to include from header. Or add a `// TODO move to header`

File: compiler/luci-interpreter/src/loader/GraphLoader.cpp
Comment (m-bronnikov) [inline compiler/luci-interpreter/src/loader/GraphLoader.cpp:210]: Since this place has not small (4 for now) number of cases with `dynamic_cast` usage I think better to rewrite this place with `switch(opcode)-case` construction in order increase performance and reduce code duplication.

Comment (m-bronnikov) [reply compiler/luci-interpreter/src/loader/GraphLoader.cpp:210]: P.S. In another PR.

+cc @binarman

File: compiler/luci-interpreter/src/loader/GraphLoader.cpp
Comment (m-bronnikov) [reply compiler/luci-interpreter/src/loader/GraphLoader.cpp:86]: I'd prefer second option for now.

**Review by seanshpark — APPROVED** [review]
LGTM

**Comment by seanshpark** [comment]
Maybe another reviewer like @binarman ?

**Comment by m-bronnikov** [comment]
TODO comment added.

**Review by seanshpark — DISMISSED** [review]
LGTM thank you!

**Comment by m-bronnikov** [comment]
> Maybe another reviewer like @binarman ?

It would be nice. I will add @BalyshevArtem and @binarman  to rewievers. I think a review of one of them will be enough.

File: compiler/luci-interpreter/src/loader/GraphLoader.cpp
Comment (m-bronnikov) [inline compiler/luci-interpreter/src/loader/GraphLoader.cpp:127]: CI fails due to implicit fallthrough here =(
Let's make explicit!

Comment (m-bronnikov) [reply compiler/luci-interpreter/src/loader/GraphLoader.cpp:129]: done!

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by BalyshevArtem — APPROVED** [review]
LGTM, thank you!

**Comment by m-bronnikov** [comment]
@seanshpark I think 2 review should be enough =)

**Pull request merged**
